mlflow/utils/oss_utils.py

```diff
@@ -0,0 +1,21 @@
+import urllib.parse
+
+from mlflow.utils.databricks_utils import get_databricks_host_creds
+from mlflow.utils.rest_utils import MlflowHostCreds
+from mlflow.utils.uri import (
+    _DATABRICKS_UNITY_CATALOG_SCHEME,
+)
+
+
+def get_oss_host_creds(server_uri=None):
+    parsed_uri = urllib.parse.urlparse(server_uri)
+    new_uri = parsed_uri.path
+    new_parsed_uri = urllib.parse.urlparse(new_uri)
+    if parsed_uri.scheme == "uc":
+        if parsed_uri.path == _DATABRICKS_UNITY_CATALOG_SCHEME:
```

Comment:
Let's add a comment here explaining that the registry URI of `uc:databricks-uc` will force mlflow to communicate with the DB UC service using the OSS pathway you have defined.

Reply:
+1 to a pydoc string for this function and to remember to mention this behavior very clearly in the documentation with an example of using this pass-through functionality

Reply:
I think I'm a little conflicted on if we expose this option to users because the UC OSS client experience on databricks would always be worse than the DB UC client pathway since UC OSS will always be trailing.  For instance, if they did this, I think they would lose all lineage info currently since we have no lineage in UC OSS yet.  We added this in as more of a convenience for us to test/validate.  Dunno if we have the notion of public but undocumented in MLflow tho as that would be the best option for it i think.

Reply:
It's also limited in functionality, right? Half of the APIs are not implemented. We do certainly have undocumented functionality in MLflow (some purely for testing purposes, such as this).